### PR TITLE
fix(node): implement TLSSocket._start

### PR DESCRIPTION
**Pull request by littledivy**
Closes https://github.com/denoland/deno/issues/19983
Closes https://github.com/denoland/deno/issues/18303
Closes https://github.com/denoland/deno/issues/16681
Closes https://github.com/denoland/deno/issues/19978

### CODE REVIEW

File: ext/node/polyfills/internal_binding/stream_wrap.ts [hunk 1]
```diff
@@ -314,9 +314,14 @@ export class LibuvStreamWrap extends HandleWrap {
     let buf = BUF;
 
     let nread: number | null;
+    const ridBefore = this[kStreamBaseField]!.rid;
     try {
       nread = await this[kStreamBaseField]!.read(buf);
     } catch (e) {
+      if (ridBefore != this[kStreamBaseField]!.rid) {
+        return this.#read();
+      }
```
Comment (bartlomieju) [inline ext/node/polyfills/internal_binding/stream_wrap.ts:323]: Add a comment explaining what is going here

File: ext/node/polyfills/internal_binding/stream_wrap.ts [hunk 2]
```diff
@@ -365,15 +370,21 @@ export class LibuvStreamWrap extends HandleWrap {
...
       while (nwritten < data.length) {
         nwritten += await this[kStreamBaseField]!.write(
           data.subarray(nwritten),
         );
       }
     } catch (e) {
+      if (ridBefore != this[kStreamBaseField]!.rid) {
+        return this.#write(req, data.subarray(nwritten));
+      }
+
```
Comment (bartlomieju) [inline ext/node/polyfills/internal_binding/stream_wrap.ts:387]: Ditto

File: ext/node/polyfills/net.ts
Comment (bartlomieju) [inline ext/node/polyfills/net.ts:1]: Revert

File: ext/node/polyfills/_stream.mjs
Comment (bartlomieju) [inline ext/node/polyfills/_stream.mjs:1]: Revert

File: client.mjs
Comment (bartlomieju) [inline client.mjs:1]: Read back the data so the program exits cleanly here. Also this can be turned into a unit test

**Review by bartlomieju — APPROVED** [review]
LGTM, awesome fix!

**Comment by dehrhard** [comment]
Yesssss! I've been waiting for that! Thank you :heart:

**Pull request merged**